parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetOutputFormat.java

```diff
@@ -317,6 +331,26 @@ private static int getMaxPaddingSize(Configuration conf) {
     return conf.getInt(MAX_PADDING_BYTES, DEFAULT_MAX_PADDING_SIZE);
   }
 
+  public static ValuesWriterFactory getValuesWriterFactory(Configuration conf) {
+    Class<? extends ValuesWriterFactory> factoryOverride =
+      (Class<? extends ValuesWriterFactory>) ConfigurationUtil.getClassFromConfig(conf, WRITER_FACTORY_OVERRIDE, ValuesWriterFactory.class);
+    if (factoryOverride == null) {
+      // no override specified, fall back to default
+      factoryOverride = ParquetProperties.DEFAULT_VALUES_WRITER_FACTORY;
+    }
+
+    try {
+      ValuesWriterFactory factory = factoryOverride.newInstance();
+      if (factory instanceof ConfigurableFactory) {
+        ConfigurableFactory configurableFactory = (ConfigurableFactory) factory;
+        configurableFactory.setConfiguration(conf);
+      }
+
+      return factory;
+    } catch (Exception e) {
```

Comment:
catching Exception is probably too much, maybe catch RuntimeException and any expected checked exceptions? Exception includes InterruptedException and other "you should not catch these" exceptions.